### PR TITLE
【code format check upgrade】 step2：cmake-format

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

Others
### PR changes

Others
### Describe

 add cmake-format to help format cmake files.
some points:

- wont format comment, because most comment are designed carefully and already have a good style. Only few of them has problem of line width over limit, which is easy to fix manully.
- use .cmake-format.py as config file, and add some pattens to tell cmake-format how to format the self-defining function such as cc_library, which can make the code more elegant and easy to read. For example:

before:
image

after:
 image

- exclude one file `paddle/fluid/operators/CMakeLists.txt`, because tools/remove_grad_op_and_kernel.py need to update it, and errors occured after format this file, so just let it keep original format.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -55,3 +55,11 @@ repos:
             (?x)^(
                 paddle/utils/.*
             )$
+-   repo: https://github.com/cheshirekow/cmake-format-precommit
+    rev: v0.6.13
+    hooks:
+    -   id: cmake-format
+        exclude: |
+            (?x)^(
+                paddle/fluid/operators/CMakeLists.txt
```
Comment (luotao1) [inline .pre-commit-config.yaml:66]: 请问这个exclude是有什么考虑么？后面会移走么？

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 #
 # Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License
```
Comment (luotao1) [inline CMakeLists.txt]: 对 License的修改有点问题。

File: cmake/cblas.cmake [hunk 3]
```diff
@@ -1,26 +1,25 @@
...
 # Find the CBlas and lapack libraries
 #
 # It will search MKLML, OpenBlas, reference-cblas, extern-openblas in order.
 #
 # If any cblas implementation found, the following variable will be set.
-#    CBLAS_PROVIDER  # one of MKLML, OPENBLAS, REFERENCE
-#    CBLAS_INC_DIR   # the include directory for cblas.
-#    CBLAS_LIBS      # a list of libraries should be linked by paddle.
-#                    # Each library should be full path to object file.
+# CBLAS_PROVIDER  # one of MKLML, OPENBLAS, REFERENCE CBLAS_INC_DIR   # the
```
Comment (luotao1) [inline cmake/cblas.cmake]: 这儿注释部分也乱了

File: .pre-commit-config.yaml (hunk 1 above)
Comment (betterpig) [reply .pre-commit-config.yaml:66]: 该文件格式化之后inference流水线[报错](https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/5771269/job/15433936)，原因为tools/remove_grad_op_and_kernel.py需要对该文件进行修改，格式化后读取该文件出错。
报错截图：
image

tools/remove_grad_op_and_kernel.py代码截图：
image

File: CMakeLists.txt (hunk 2 above)
Comment (betterpig) [reply CMakeLists.txt]: 考虑到注释一般都是已经排好版的，只要极少数注释会具有列宽超限问题，因此决定不对注释进行格式化，对于注释的把控主要靠开发同学自身，以及cmakelint对于列宽超限提示的辅助。
已通过在.cmake-format.py里设置 `enable_markup = False` 屏蔽掉对注释的格式化。

File: cmake/cblas.cmake (hunk 3 above)
Comment (betterpig) [reply cmake/cblas.cmake]: 见上一条评论。

File: .pre-commit-config.yaml (hunk 1 above)
Comment (luotao1) [reply .pre-commit-config.yaml:66]: 明白了，remove_grad_op_and_kernel.py 是推理用来裁剪反向Op的，可以后续看下如何格式化

**Review by luotao1 — APPROVED** [review]
LGTM

**Comment by luotao1** [comment]
2022-06-03 11:39:23 3. You must be approved by chenwhql or zyfncg for paddle/phi/kernels/declarations.h using. Thanks!
https://xly.bce.baidu.com/paddlepaddle/paddle/newipipe/detail/5776637/job/15453992
@chenwhql 请问这个PR是怎么触发这个规则的呢，要如何修复？

**Review by chenwhql — APPROVED** [review]

**Pull request merged**